tests/pattern_rewriter/test_pattern_rewriter.py

```diff
@@ -245,7 +244,7 @@ def test_recursive_rewriter():
     class Rewrite(RewritePattern):
         @op_type_rewrite_pattern
         def match_and_rewrite(self, op: Constant, rewriter: PatternRewriter):
-            if not isa(op_val := op.value, IntegerAttr):
+            if not isinstance(op_val := op.value, IntegerAttr):
```

Comment:
What happens in those cases? `isa` is returning an error, or is it just that `isinstance` can do it so we should follow that?

Reply:
```
Argument type is partially unknown
  Argument corresponds to parameter "hint" in function "isa"
  Argument type is "TypeForm[IntegerAttr[Unknown]]"
```

Reply:
I assume that this isn't the case for newer versions of Python, where generic arguments can have default values, where `IntegerAttr` would be shorthand for `IntegerAttr[IntegerType | IndexType]`